### ISSUE TITLE
segmentation fault on total(,/double)

### ISSUE THREAD

**Issue by brandy125** [gnudatalanguage/gdl#680]
GDL> a=total(exp(dcomplex(0,dindgen(100))),/double)
Magick: abort due to signal 11 (SIGSEGV) "Segmentation Fault"...
Aborted (core dumped)

**Comment by maynardGK** [comment]
Here's some more info on it. From Windows' build 1163 release/64 bit
````
GDL> hash(!gdl)
EPOCH:  1576310400
RELEASE:1.0.0-rc.1 git
GDL_POSIX:   1
BUILD_DATE:Dec 14 2019
GDL_USE_WX:   0
MAP_QUALITY:
CRUDE
GDL_NO_DSFMT:   0
GDL> a=total(exp(dcomplex(0,dindgen(100))),/double)
Assertion failed!

Program: D:\f\gdl64\install\gdl\bin\gdl.exe
File: C:/projects/gdl/src/gdlarray.hpp, Line 185

Expression: ix < sz
Magick: abort due to signal 22 (SIGABRT) "Abort"...
````
 Previous build 1104 also has this error (Oct 12)
But 1044 does not (#635 )

````
GDL> hash(!gdl)
EPOCH:  1569222000
RELEASE:0.9.9 git
GDL_POSIX:   0
BUILD_DATE:Sep 23 2019
GDL_USE_WX:   0
MAP_QUALITY:CRUDE

GDL_NO_DSFMT:   0
GDL> a=total(exp(dcomplex(0,dindgen(100))),/double)
GDL>
````
fails upon update making build 1101 from https://github.com/gnudatalanguage/gdl/commit/fe55e092a0772a26d89842eab790194aed55c923
(see https://github.com/gnudatalanguage/gdl/pull/653)
````
GDL> hash(!gdl)
EPOCH:  1570777200
RELEASE:0.9.9 git
BUILD_DATE:Oct 11 2019
GDL_USE_WX:   0
MAP_QUALITY:CRUDE
GDL_NO_DSFMT:
   0
GDL> a=total(exp(dcomplex(0,dindgen(100))),/double)
Assertion failed!

Program: D:\f\gdl64\install\gdl\bin\gdl.exe
File: C:/projects/gdl/src/gdlarray.hpp, Line 185

Expression: ix < sz
Magick: abort due to signal 22 (SIGABRT) "Abort"...
````

**Comment by maynardGK** [comment]
Another factoid: take away the /DOUBLE keyword and the appropriate behavior seems to hold:

````
GDL> a=total(exp(dcomplex(0,dindgen(100))))
GDL> help,a
A               DCOMPLEX  = (     -0.39460748,      0.37919463)
GDL> hash(!gdl)
EPOCH:  1572940800
RELEASE:0.9.9 git
GDL_POSIX:   1
BUILD_DATE:Nov  5 2019
GDL_USE_WX:   0
MAP_QUALITY:CRUDE

GDL_NO_DSFMT:   0
````

also,
````
GDL> theta=2*!Dpi*Dindgen(100)/100.
GDL> ee=exp(dcomplex(0,theta))
GDL> a=total(ee)
GDL> print,a
( -7.9936058e-015,  5.2319260e-015)
````

**Comment by GillesDuvert** [comment]
A typo in the code. Not unexpected, given the circumstances... Cured and soon patched in GH.

**Comment by GillesDuvert** [comment]
solved by PR #684
